SECEdgar/base.py

```diff
@@ -15,22 +17,17 @@ class _EDGARBase(ABC):
     """Abstract Base Class for EDGAR requests.
 
     Attributes:
-        retry_count (int, optional): Desired number of retries if a request fails.
-            Defaults to 3.
-        pause (float, optional): Pause time between retry attempts.
-            Defaults to 0.5.
-        count (int, optional): Number of reports to fetch. Defaults to 10.
-            Will fetch all if total available is less than count.
+        client (SECEdgar.network_client.NetworkClient) : NetworkClient object that handles
+            getting data from EDGAR.
+        url (str): URL endpoint.
+        params (dict): Dictionary of parameters to add.
 
     .. versionadded:: 0.1.5
     """
-    _BASE = "http://www.sec.gov/cgi-bin/"
 
     def __init__(self, **kwargs):
-        self.retry_count = kwargs.get("retry_count", 3)
-        self.pause = kwargs.get("pause", 0.5)
-        self.count = kwargs.get("count", 10)
-        self._params = dict()
+        self._client = NetworkClient(**kwargs)
```

Comment:
`ABC` should not have implementation details like above. 

it should be just a class and whosoever conforms to it, need to implement the respective methods.
```
class SECEdgarBase(object):
    __metaclass__ = ABC
    
    @abc.abstractmethod
    def __init__(self, networkClient, count):
        return
```

Reply:
Should we move these details to the `NetworkClient` class?